weatherman.py

```diff
@@ -0,0 +1,164 @@
+import os
+import argparse
+from datetime import datetime
+
+
+RED = "\033[31m"
+BLUE = "\033[34m"
+RESET = "\033[0m"
+
+
+class WeatherReading:
+    def __init__(self, date, max_temp, min_temp, max_humidity, mean_humidity):
+        self.date = date
+        self.max_temp = max_temp
+        self.min_temp = min_temp
+        self.max_humidity = max_humidity
+        self.mean_humidity = mean_humidity
+
+    def __repr__(self):
+        return (f"WeatherReading(date={self.date}, max_temp={self.max_temp}, "
+                f"min_temp={self.min_temp}, max_humidity={self.max_humidity}, "
+                f"mean_humidity={self.mean_humidity})")
+
+
+def parse_weather_file(file_path):
+    readings = []
+    try:
+        with open(file_path, 'r') as file:
```

Comment:
The with statement is used to code more elegantly and precisely. Instead of explicitly closing the file in a finally block, we can simply use with as it ensures that the file is automatically closed after our usage.

Reply:
You need to explore it well. You are unclear on the right usage of context manager.